### ISSUE TITLE
Foxx: Repository.byExample vs Repository.firstExample

### ISSUE THREAD

**Issue by armatys** [arangodb/arangodb#1323]
Hello,
I encountered an issue, while developing a Foxx app. The observed behavior seems incorrect, but forgive me if it's not, as I've started using Arango just today.

```
// Some setup..
var users = require('repositories/users').Repository;
var User = require('models/User').Model;
var Users = new users(applicationContext.collection('users'), {
    model: User
});

var email = 'some_nonexistent@email.com';
var userList = Users.byExample({email: email});
var user = Users.firstExample({email: email});
```
The issue is when I try to find a user which does not exist in the databse. The first method `byExample` correctly returns a list with 0 element, but `firstExample` method actually returns an object - it has some default attributes set. Is this on purpose? It would be inconsistent with `Users.collection.firstExample`, which actually returns `null`.

Tested on arango 2.5.2, OS X

**Comment by pluma** [comment]
I agree that it shouldn't return anything (i.e. `null`), but I'm sure changing that behaviour would be a backwards-incompatible change and break some code.

+1 on fixing this. @fceller there's no way to gracefully introduce a change here, can we just be bold and fix this in 2.6?

**Comment by fceller** [comment]
I agree that "firstExample" should behave like the collection version and return "null".

If we fix this, we should introduce it only in 2.6 (not the 2.5 branch).
